### PR TITLE
add logger module

### PR DESCRIPTION
**Pull request by Xu-Kai**

### CODE REVIEW

File: scalableml/scheduler/executor_manager.py [hunk 1]
```diff
@@ -409,12 +410,21 @@ def start_service(node_uid, headip, headport, password, nodeip, server_port, obj
         nodeip(str): current node ip
         server_port(str): the port of executor manager of current node
         obj_port(str): ip and port of object store of current node
+        log_file(str): log file name 
     """
-    serve(node_uid, headip, headport, password, nodeip, server_port, obj_port)
+
+    while logger.hasHandlers():
+            logger.removeHandler(logger.handlers[0])
```
Comment (gkevinyen5418) [inline scalableml/scheduler/executor_manager.py:417]: The indentation looks weird.

File: scalableml/start_worker.py [hunk 2]
```diff
@@ -34,16 +30,33 @@ def main():
...
     parser.add_argument("--headport", nargs="?", default="6379", type=int, help="head ip port to use (default : 6379)")
     parser.add_argument("--passwd", nargs="?", default="", type=str, help="head redis password")
+    parser.add_argument("--logpath", nargs="?", default="~/.local/scalableml", type=str, help="head redis password")
     args = parser.parse_args()
 
+    time_str = time.strftime("%Y-%m-%d-%H-%M-%S")
+
+    directory = os.path.expanduser("~/.local/scalableml") + "/" + time_str
+    if not os.path.exists(directory):
+            os.makedirs(directory)
```
Comment (gkevinyen5418) [inline scalableml/start_worker.py:40]: indentation

File: scalableml/start_worker.py [hunk 3]
```diff
@@ -34,16 +30,33 @@ def main():
     parser.add_argument("--headip", nargs="?", default="127.0.0.1", type=str, help="head ip to use")
     parser.add_argument("--headport", nargs="?", default="6379", type=int, help="head ip port to use (default : 6379)")
     parser.add_argument("--passwd", nargs="?", default="", type=str, help="head redis password")
+    parser.add_argument("--logpath", nargs="?", default="~/.local/scalableml", type=str, help="head redis password")
```
Comment (gkevinyen5418) [inline scalableml/start_worker.py:33]: The help message needs to be corrected.

File: scalableml/start_worker.py [hunk 4]
```diff
@@ -34,16 +30,33 @@ def main():
...
     args = parser.parse_args()
 
+    time_str = time.strftime("%Y-%m-%d-%H-%M-%S")
+
+    directory = os.path.expanduser("~/.local/scalableml") + "/" + time_str
```
Comment (gkevinyen5418) [inline scalableml/start_worker.py:38]: Should use `args.logpath`

Comment (Xu-Kai) [reply scalableml/start_worker.py]: I have changed the path.

File: scalableml/start_worker.py (hunk 3 above)
Comment (Xu-Kai) [reply scalableml/start_worker.py]: I have changed the help message.

File: scalableml/scheduler/executor_manager.py (hunk 1 above)
Comment (Xu-Kai) [reply scalableml/scheduler/executor_manager.py]: Correct the indentation.

File: scalableml/start_worker.py [hunk 5]
```diff
@@ -30,12 +30,12 @@ def main():
...
-    parser.add_argument("--logpath", nargs="?", default="~/.local/scalableml", type=str, help="head redis password")
+    parser.add_argument("--logpath", nargs="?", default="~/.local/scalableml", type=str, help="log file directory")
     args = parser.parse_args()
 
     time_str = time.strftime("%Y-%m-%d-%H-%M-%S")
 
-    directory = os.path.expanduser("~/.local/scalableml") + "/" + time_str
+    directory = os.path.expanduser(args.logpath) + "/" + time_str
     if not os.path.exists(directory):
             os.makedirs(directory)
```
Comment (gkevinyen5418) [inline scalableml/start_worker.py:40]: Indentation

Comment (Xu-Kai) [reply scalableml/start_worker.py]: Done

**Review by gkevinyen5418 — APPROVED** [review]
LGTM. I am also closing this so we can switch to our improved new workflow.

**Pull request merged**